### PR TITLE
Use ScreenCoordsXY for window functions

### PR DESCRIPTION
**Pull request by tupaschoal**
I got a bit excited, you might want to review it commit by commit, they are atomic and all of them compile individually and combined :)

### CODE REVIEW

**Comment by Gymnasiast** [comment]
Could you rebase this PR?

**Comment by tupaschoal** [comment]
Done rebasing.

File: src/openrct2/interface/Window.cpp [hunk 1]
```diff
@@ -1239,26 +1241,26 @@ void window_draw_viewport(rct_drawpixelinfo* dpi, rct_window* w)
...
 
-void window_set_position(rct_window* w, int32_t x, int32_t y)
+void window_set_position(rct_window* w, ScreenCoordsXY screenCoords)
 {
-    window_move_position(w, x - w->x, y - w->y);
+    window_move_position(w, ScreenCoordsXY(screenCoords.x - w->x, screenCoords.y - w->y));
 }
 
-void window_move_position(rct_window* w, int32_t dx, int32_t dy)
+void window_move_position(rct_window* w, ScreenCoordsXY screenCoords)
```
Comment (duncanspumpkin) [inline src/openrct2/interface/Window.cpp]: should make this clear that screenCoords is a delta value. Suggest renaming this `deltaCoords`.

Comment (tupaschoal) [reply src/openrct2/interface/Window.cpp]: Done!

**Pull request merged**